src/function.ts

```diff
@@ -30,11 +38,50 @@ export type AnyLambda = Function<any, any>;
  */
 export class Function<P, O> {
   readonly kind = "Function" as const;
+  public static readonly FunctionlessType = "Function";
+  readonly resource: aws_lambda.IFunction;
+
+  static promises: Promise<any>[] = [];
```

Comment:
Should we make sure this is a global singleton? Or do we want to rely on the `functionless` module being a singleton because of `peerDeps`? 

My preferred method is with a symbol:

```ts
const PromisesSymbol = Symbol.for("functionless.Function.promises");

public static readonly promises = (global[PromisesSymbol] = global[PromisesSymbol] ?? []);
```

Reply:
Good idea.